tests/functionaltests/UXTestHelpers.mm

```diff
@@ -106,4 +114,27 @@
     }
 }
 
+//
+// UXEvent methods
+//
+void UXEvent::Signal() {
+    [_condition lock];
+
+    _signaled = true;
+    [_condition signal];
+
+    [_condition unlock];
+}
+
+bool UXEvent::Wait(int timeOutInSeconds) {
```

Comment:
I have no problem changing the API to be WaitFor(....) which returns true if we timed out or false if we were signaled. Incidentally, the iOS documentation for its [NSCondition waitUntilDate:...] is confusing. It returns YES if the condition was signaled. My instincts would expect [NSCondition waitUntilDate:..] should return YES if it waited the full period.

---
In reply to: [98269735](https://github.com/Microsoft/WinObjC/pull/1855#discussion_r98269735) [](ancestors = 98269735,98268635,98267209)